Management/src/ActuatorWeb/OpenTelemetryExtensions.cs

```diff
@@ -0,0 +1,73 @@
+using System.Text.RegularExpressions;
+using OpenTelemetry;
+using OpenTelemetry.Context.Propagation;
+using OpenTelemetry.Instrumentation.AspNetCore;
+using OpenTelemetry.Instrumentation.Http;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Steeltoe.Common;
+using B3Propagator = OpenTelemetry.Extensions.Propagators.B3Propagator;
+
+namespace Steeltoe.Samples.ActuatorWeb;
+
+internal static class OpenTelemetryExtensions
+{
+    private const string DefaultEgressIgnorePattern = "/api/v2/spans|/v2/apps/.*/permissions";
+
+    public static void ConfigureOpenTelemetry(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.AddOpenTelemetry().WithMetrics(metrics =>
+        {
+            metrics.AddAspNetCoreInstrumentation().AddHttpClientInstrumentation().AddRuntimeInstrumentation();
+        }).WithTracing(tracing =>
+        {
+            tracing.AddAspNetCoreInstrumentation().AddHttpClientInstrumentation();
+
+            string? zipkinExporterAddress = configuration.GetValue<string>("OTEL_EXPORTER_ZIPKIN_ENDPOINT");
+
+            if (!string.IsNullOrEmpty(zipkinExporterAddress))
+            {
+                tracing.AddZipkinExporter(zipkinExporterOptions => zipkinExporterOptions.Endpoint = new Uri(zipkinExporterAddress));
+            }
+        });
+
+        services.ConfigureOpenTelemetryTracerProvider((serviceProvider, tracerProviderBuilder) =>
+        {
+            var appInfo = serviceProvider.GetRequiredService<IApplicationInstanceInfo>();
+            tracerProviderBuilder.SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(appInfo.ApplicationName ?? "ActuatorWeb-fallback"));
```

Comment:
We should assume that `appInfo.ApplicationName` is not null, so a `!` suppression can be used.
If that's not the case somehow, the code should throw instead of silently continuing using a fallback value.

Reply:
Fair. I had been testing some different registration scenarios and forgot to remove this... I don't think I ever actually hit the fallback value